fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/FedoraToOCFLObjectIndexUtilImpl.java

```diff
@@ -80,8 +83,11 @@ public void rebuild() {
                                 fedoraIds.add(headers.getId());
                                 if (headers.isArchivalGroup()) {
                                     rootId.set(headers.getId());
+                                } else if (Objects.equals(headers.getInteractionModel(),
+                                        PERSISTENCE_FEDORA_NON_RDF_SOURCE_DESCRIPTION)) {
+                                    rootId.set(headers.getParent());
```

Comment:
I'm trying to understand if this is the case and I can't test at the moment, so I may be totally off, but it seems like if you have an archival group which contains a binary, then everything else in the AG would also have the binary as its root instead of the AG? Do we only want to set the root for a description to the binary if its not in an AG, or do we always want to do this?

Reply:
I think that's right @bbpennel.  And we also want to account for fcr:acl  as well.

Reply:
Maybe changing this to
```
} else if (Objects.equals(headers.getInteractionModel(),
                                        FEDORA_NON_RDF_SOURCE_DESCRIPTION_URI) &&
               rootId.get() == null) {
```
So it only sets it if the archival group hasn't set it, and if the archival group comes later it would overwrite.

Reply:
Perhaps this is my lack of knowledge of how this was designed. In the case of an AG we know the root, but in the case of something else what (in the JSON) identifies it as the root?

Reply:
hmm, i don't think anything in the json identifies it as the root. We had talked about potentially using a special name for the root object, although i'm forgetting what it was at the moment. Seems like once we find the root, we don't need to hunt for roots anymore, so maybe only do any of `if (headers.isArchivalGroup()) {` block if the root id is still null?

Reply:
Right, my question is what is the root ID for a non-archival group? I'd prefer not to check for each Fedora type, so if we can codify this in the JSON all the better.

Before this PR the root for the binary and binary description was `null` because the code was only setting a root for AGs or if there was a single file it became its own root.

We need a way for multiple files in a single object (binary description, ACL, etc) but not an AG to have the root defined. Maybe on top of `"archivalGroup":(true|false)` we need a `"objectRoot":(true|false)`?

Reply:
For a non-archival group,  the root is itself. 
http://localhost:8080/rest/binary ->   root= info:fedora/binary  resource=info:fedora/binary
http://localhost:8080/rest/binary ->   root=  info:fedora/binary resource= info:fedora/binary/fcr:metadata

Reply:
Perhaps I'm missing something?

Reply:
What I am wondering is is there a way to determine the correct root Id for resources without checking if 
```
if (
  Objects.equals(obj.getInteractionModel(), FEDORA_NON_RDF_SOURCE_DESCRIPTION_URI) ||
  obj.getId().endsWith(FCR_ACL))
) {
```

Otherwise what I am doing will (as in your PR) have to be expanded to cover a bunch of different cases. If we can generalize that relationship between supporting resources (like binary descriptions and ACLs) and the resources they support, then we don't need to check the type of object in the rebuild process.

Reply:
I think I see what you're saying.   If we add "objectRoot": (true|false) to the ResourceHeaders.json, then we don't have to deal with inspecting the interaction model.